### PR TITLE
Extract API bundle and add Baseline plugin

### PR DESCRIPTION
**Pull request by royteeuwen**

### CODE REVIEW

**Comment by royteeuwen** [comment]
Will solve https://github.com/orbinson/aem-groovy-console/issues/28

File: api/src/main/groovy/be/orbinson/aem/groovy/console/api/context/ServletScriptContext.groovy [hunk 1]
```diff
@@ -2,10 +2,14 @@ package be.orbinson.aem.groovy.console.api.context
 
 import org.apache.sling.api.SlingHttpServletRequest
 import org.apache.sling.api.SlingHttpServletResponse
+import org.osgi.annotation.versioning.ConsumerType
+
+import java.util.function.Consumer
```
Comment (bdhoine) [inline api/src/main/groovy/be/orbinson/aem/groovy/console/api/context/ServletScriptContext.groovy:7]: should this import be added?

File: api/src/main/groovy/be/orbinson/aem/groovy/console/api/context/impl/package-info.groovy [hunk 2]
```diff
@@ -0,0 +1,3 @@
+@org.osgi.annotation.bundle.Export
+@org.osgi.annotation.versioning.Version("1.0.0")
```
Comment (bdhoine) [inline api/src/main/groovy/be/orbinson/aem/groovy/console/api/context/impl/package-info.groovy:2]: should this be inline with the maven version as we will only bump if needed?

File: api/src/main/groovy/be/orbinson/aem/groovy/console/notification/EmailNotificationService.groovy [hunk 3]
```diff
@@ -1,10 +1,13 @@
 package be.orbinson.aem.groovy.console.notification
 
 import be.orbinson.aem.groovy.console.response.RunScriptResponse
+import org.osgi.annotation.versioning.ConsumerType
+import org.osgi.annotation.versioning.ProviderType
```
Comment (bdhoine) [inline api/src/main/groovy/be/orbinson/aem/groovy/console/notification/EmailNotificationService.groovy:5]: ProviderType not used

**Review by bdhoine — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 4]
```diff
@@ -7,6 +7,11 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+### Changed
+
+- Split the bundle into an API and Core bundle and add package versions.
```
Comment (bdhoine) [inline CHANGELOG.md:12]:
```suggestion
- Split the bundle into an API and Core bundle and add package versions: [#31](https://github.com/orbinson/aem-groovy-console/issues/31)
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -7,6 +7,11 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+### Changed
+
+- Split the bundle into an API and Core bundle and add package versions. 
+- Use the bnd-baseline-maven-plugin to verify when to upgrade the package versions
```
Comment (bdhoine) [inline CHANGELOG.md:13]:
```suggestion
- Use the bnd-baseline-maven-plugin to verify when to upgrade the package versions: [#31](https://github.com/orbinson/aem-groovy-console/issues/31)
```

**Review by bdhoine — APPROVED** [review]

**Pull request merged**